Source/SA/RFmx Instr/RFmxInstr.cs

```diff
@@ -0,0 +1,112 @@
+{
+    public static class RFmxInstr
+    {
+        #region Type Definitions
+        public struct InstrumentConfiguration
+        {
+            public LocalOscillatorSharingMode LOSharingMode;
+            public LocalOscillatorRoutingConfiguration[] LORoutingConfigurations;
+            public LocalOscillatorFrequencyOffsetConfiguration LOOffsetConfiguration;
+
+            public static InstrumentConfiguration GetDefault()
+            {
+                return new InstrumentConfiguration
+                {
+                    LOSharingMode = LocalOscillatorSharingMode.Automatic,
+                    LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[] { LocalOscillatorRoutingConfiguration.GetDefault() },
+                    LOOffsetConfiguration = LocalOscillatorFrequencyOffsetConfiguration.GetDefault()
+                };
+            }
+
+            public static InstrumentConfiguration GetDefault(RFmxInstrMX sessionHandle)
+            {
+                InstrumentConfiguration instrConfig = GetDefault(); // covers case for sub6 instruments with a single configurable LO
+                // lo routing configurations will now be overridden if the instrument has number of LOs != 1
+                sessionHandle.GetInstrumentModel("", out string instrumentModel);
+                if (instrumentModel.Equals("NI PXIe-5830"))
+                    instrConfig.LORoutingConfigurations[0].ChannelName = "LO2";
+                else if (Regex.IsMatch(instrumentModel, "NI PXIe-5831"))
+                {
+                    LocalOscillatorRoutingConfiguration lo1RoutingConfig = LocalOscillatorRoutingConfiguration.GetDefault();
+                    lo1RoutingConfig.ChannelName = "LO1";
+                    LocalOscillatorRoutingConfiguration lo2RoutingConfig = LocalOscillatorRoutingConfiguration.GetDefault();
+                    lo2RoutingConfig.ChannelName = "LO2";
+                    instrConfig.LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[] { lo1RoutingConfig, lo2RoutingConfig };
+                }
+                else if (Regex.IsMatch(instrumentModel, "NI PXIe-5(82.|645R)")) // matches on any baseband instrument without an LO
+                {
+                    instrConfig.LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[0]; // baseband instruments have no LOs
+                }
+                return instrConfig;
+            }
+        }
+        #endregion
+
+        #region Instrument Configurations
+        public static void ConfigureInstrument(RFmxInstrMX instrHandle, InstrumentConfiguration instrConfig)
+        {
+            instrHandle.GetInstrumentModel("", out string instrumentModel);
+
+            if (Regex.IsMatch(instrumentModel, "NI PXIe-5(82.|645R)")) // matches on any baseband instrument without an LO
+                return; // return early since the instrument doesn't have any LOs that can be configured

```

Comment:
My preference here would instead be first do the check on line 59 to see if we are using a supported non-baseband VST. For all other instruments I would like to return. This is because I want this to be able to work on a standalone VSA like a 5668. Functionally, that happens at present because execution would skip all of the LO configs since the 5668 wouldn't match the string. But, I think we could make this clearer and simpler by just moving the line on 59 first and otherwise exiting.

Reply:
i.e. currently in LabVIEW I have the following: 
![image](https://user-images.githubusercontent.com/47825680/67804047-3125b200-fa5c-11e9-8a96-2be295d2ebc6.png)

Reply:
i just made some changes to the code to remove the case structure you have there. i don't see why what we have wouldn't work on a 5668.

Reply:
Because none of these properties are supported on that instrument. All will cause errors if called.

Reply:
Ah yes of course my bad. Well i think based on our convo yesterday some of this might change based on our decision this morning on the fate of the lo sharing stuff. Agreed that we can return early for unsupported instruments.

Reply:
Agreed. Will mostly be a moot point, though not entirely since I think the Auto property for SG SA shared will likely still error on unsupported devices. Let's hold off until later today.